src/wallet/rpcsigner.cpp

```diff
@@ -0,0 +1,594 @@
+        throw JSONRPCError(RPC_WALLET_ERROR, e.what());
+    }
+}
+
+static UniValue enumeratesigners(const JSONRPCRequest& request)
+{
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    CWallet* const pwallet = wallet.get();
+
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() != 0) {
+        throw std::runtime_error(
+            RPCHelpMan{"enumeratesigners\n",
+                "Returns a list of external signers from -signer and associates them\n"
+                "with the wallet until you stop bitcoind.\n",
+                {},
+                RPCResult{
+                    "{\n"
+                    "  \"signers\" : [                              (json array of objects)\n"
+                    "    {\n"
+                    "      \"masterkeyfingerprint\" : \"fingerprint\" (string) Master key fingerprint\n"
+                    "    }\n"
+                    "    ,...\n"
+                    "  ]\n"
+                    "}\n"
+                },
+                RPCExamples{""}
+            }.ToString()
+        );
+    }
+
+    UniValue signers = getsigners(pwallet);
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("signers", signers);
+    return result;
+}
+
+ExternalSigner *GetSignerForJSONRPCRequest(const JSONRPCRequest& request, int index, CWallet* pwallet) {
+    if (pwallet->m_external_signers.empty()) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "First call enumeratesigners");
+    }
+
+    // If no fingerprint is specified, return the only available signer
+    if (request.params.size() < size_t(index + 1) || request.params[index].isNull()) {
+        if (pwallet->m_external_signers.size() > 1) {
+            throw JSONRPCError(RPC_WALLET_ERROR, "Multiple signers found, please specify which to use");
+        }
+        return &pwallet->m_external_signers.front();
+    }
+
+    const std::string fingerprint = request.params[index].get_str();
+    for (ExternalSigner &candidate : pwallet->m_external_signers) {
+        if (candidate.m_fingerprint == fingerprint) return &candidate;
+    }
+    throw JSONRPCError(RPC_WALLET_ERROR, "Signer fingerprint not found");
+}
+
+UniValue signerdissociate(const JSONRPCRequest& request)
+{
+    std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest(request);
+    CWallet* const pwallet = wallet.get();
+
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() > 1) {
+        throw std::runtime_error(
+            RPCHelpMan{"signerdissociate",
+                "Disossociates external signer from the wallet.\n",
+                {
+                    {"fingerprint", RPCArg::Type::STR, /* opt */ true, /* default_val */ "", "Master key fingerprint of signer"},
+                },
+                RPCResult{""},
```

Comment:
The result is "null", not "". This causes the crash.

Reply:
Fixed!